app/pods/publications/publication/publishpreview/controller.js

```diff
@@ -1,19 +1,122 @@
 import Controller from '@ember/controller';
-import { action } from '@ember/object';
+import {
+  action, set
+} from '@ember/object';
+import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
+import CONFIG from 'fe-redpencil/utils/config';
+import { A } from '@ember/array';
+import moment from 'moment';
 
 export default class PublicationPublishPreviewController extends Controller {
+  // Services.
+  @service activityService;
+  @service subcasesService;
+
   // properties for making the design
   @tracked withdrawn = true;
+  @tracked testDate = new Date();
+  @tracked panelCollapsed = false;
+  @tracked showLoader = false;
+  @tracked showpublicationModal = false;
+  @tracked panelIcon = 'chevron-down'
+  @tracked publicationActivity = {
+    previewActivity: {},
+    mailContent: '',
+    subjectContent: '',
+    pieces: A([]),
+  };
 
   get publishPreviewActivities() {
     const publishPreviewActivities = this.model.publishPreviewActivities.map((activity) => activity);
     return publishPreviewActivities;
   }
 
+  /** BS PUBLICATION ACTIVITIES **/
+
   @action
-  async cancelExistingPublishPreviewActivity() {
-    alert('this action is implemented in another ticket');
+  async requestPublicationModal(activity) {
+    this.publicationActivity.pieces = activity.usedPieces;
+    const names = this.publicationActivity.pieces.map((piece) => piece.name).join('\n');
+    set(this.publicationActivity, 'previewActivity', activity);
+    set(this.publicationActivity, 'mailContent', CONFIG.mail.publishRequest.content.replace('%%attachments%%', names));
+    set(this.publicationActivity, 'mailSubject', CONFIG.mail.publishRequest.subject.replace('%%nummer%%', this.model.publicationFlow.publicationNumber));
+    this.showpublicationModal = true;
+    console.log(this.publicationActivity);
+  }
+
+  @action
+  async cancelPublicationModal() {
+    set(this.publicationActivity, 'previewActivity', {});
+    set(this.publicationActivity, 'pieces', A([]));
+    set(this.publicationActivity, 'mailContent', '');
+    set(this.publicationActivity, 'mailSubject', '');
+    this.showpublicationModal = false;
+  }
+
+  @action
+  async createPublicationActivity() {
+    this.showpublicationModal = false;
+    this.showLoader = true;
+
+    // Fetch the type.
+    const publishSubCaseType = await  this.store.findRecord('subcase-type', CONFIG.SUBCASE_TYPES.publicatieBS.id);
+
```

Comment:
Spatie teveel na de `await`

Reply:
ja raar. Heb dat er in een andere branch uitgehaald, maar hier nu ook.